### PR TITLE
Booking modal - Fabricio

### PR DESCRIPTION
**Pull request by rfabricioflores**
Added booking modal using js and sass. Works from all the pages by running the `book(title, challengeID);` function!
You can test it by running the function in console!

### CODE REVIEW

File: src/scripts/booking.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        <input type="text" name="name" id="name">
+
+        <label for="email">E-mail</label>
+        <input type="email" name="email" id="email">
+
+        <label for="time">What time?</label>
+        <select name="time" id="time"></select>
+
+        <label for="participants">How many participants?</label>
+        <select name="participants" id="participants">
```
Comment (Arash-SJPI22) [inline src/scripts/booking.js:98]: As mentioned earlier, the max and min number of participants should be loaded from the data so when you book you get the right number of participants.

File: challenges.html [hunk 2]
```diff
@@ -134,7 +134,8 @@ <h2>Follow us on</h2>
             </nav>
         </div>
     </footer>
-    <script src="app.js"></script>
+    <script src="app.js" type="module"></script>
```
Comment (Arash-SJPI22) [inline challenges.html]: why make app.js a module if you don't import src/scripts/booking.js in to it?

File: src/scripts/booking.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+  addCancelOption();
+}
+
+// Submit Step 1
+async function submitStepOne(challenge) {
+  date = document.querySelector("#bookingDate").value;
+
+  if (!date) {
+    alert('Please fill the date field!');
```
Comment (ved0) [inline src/scripts/booking.js:62]: a suggestion would be to change how the alerts are displayed!

File: src/scripts/booking.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  finalRes = data;
+  bookStepThree();
+}
+
+// Book a room (Third Step)
+function bookStepThree() {
+  const bookingPartTwoDiv = document.querySelector('#bookingPartTwo');
+  body.removeChild(bookingPartTwoDiv);
+
+  const thirdTemplate = `
```
Comment (ved0) [inline src/scripts/booking.js:175]: I like the way your HTML-template looks and is written. Nice way to do it.

File: src/scripts/booking.js (hunk 1 above)
Comment (rfabricioflores) [reply src/scripts/booking.js:98]: 👍

File: challenges.html (hunk 2 above)
Comment (rfabricioflores) [reply challenges.html]: Glömde ta bort "module"

File: src/scripts/booking.js (hunk 3 above)
Comment (rfabricioflores) [reply src/scripts/booking.js:62]: Suggestion accepted! Will do next time :D

File: src/scripts/booking.js (hunk 4 above)
Comment (rfabricioflores) [reply src/scripts/booking.js:175]: Thanks man

**Comment by rfabricioflores** [comment]
I'll fix these suggestions and issues until the next PR!

File: src/scripts/booking.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    // Time selected
+    const timeDiv = document.querySelector('#time');
+    const timeSelected = timeDiv.options[timeDiv.selectedIndex].text;
+
+    // Participants
+    let particiDiv = document.querySelector('#participants');
+    let participantSelected = particiDiv.options[particiDiv.selectedIndex].value;
+    participantSelected = parseInt(participantSelected);
+
+    submitStepTwo(challengeNumber, name,  email, date, timeSelected, participantSelected);
```
Comment (richardolsson) [inline src/scripts/booking.js:138]: To me, this code was confusing until I realized that `submitStepTwo()` doesn't just submit step two, but also proceeds to step 3. I would have moved the code to proceed from step two to step three into a separate function, or just to after this line (i.e. line 139). That way, a reader of the code can see that two things happen here: First the data from step two is submitted, and then the user is presented with step three.

**Review by richardolsson — COMMENTED** [review]
This looks nice! I added a comment on how I think the code could have been improved, see below. 👇

**Pull request merged**